### PR TITLE
logins working asynchronously

### LINKED ISSUE #245: Use `AsyncLoginsStorage` class
Currently, we use the `LoginsStorage` class from `sync-login`. Instead, we should be using `AsyncLoginsStorage`.

### PR DESCRIPTION
**Pull request by sashei**
Fixes #245
Connected to #244
## To Do

- [x] add “WIP” to the PR title if pushing up but not complete nor ready for review
- [x] double check the original issue to confirm it is fully satisfied
- [x] add testing notes and screenshots in PR description to help guide reviewers
- [x] add unit tests
  - optional: consider adding instrumentation (integration/UI) tests
- consider running this branch in a debug simulator and check for memory leak notifications or any warnings
- [x] make sure CI builds are passing (e.g.: fix lint and other errors)

### CODE REVIEW

File: app/src/main/java/mozilla/lockbox/store/DataStore.kt [hunk 1]
```diff
@@ -131,13 +139,24 @@ open class DataStore(
     }
 
     private fun updateList() {
-        this.listSubject.accept(backend.list())
+        if (!backend.isLocked()) {
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/store/DataStore.kt]: 😍

File: app/src/main/java/mozilla/lockbox/store/DataStore.kt [hunk 2]
```diff
@@ -76,53 +80,57 @@ open class DataStore(
...
+                .asSingle(Dispatchers.Default)
+                .subscribe { _, _ ->
+                    updateList()
+                }
+                .addTo(compositeDisposable)
         }
     }
 
-    fun unlock() {
+    private fun unlock() {
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/store/DataStore.kt]: Conceptually, I'd like to be able `unlock` and then be able to wait for it to finish before launching `sync`, all from within `updateCredentials`.

File: app/src/test/java/mozilla/lockbox/store/DataStoreTest.kt [hunk 3]
```diff
@@ -74,59 +78,27 @@ class DataStoreTest : DisposingTest() {
...
-    }
-
     @Test
     fun testTouch() {
-        val stateObserver = createTestObserver<State>()
-        val listObserver = createTestObserver<List<ServerPassword>>()
-
         dispatcher.dispatch(DataStoreAction.Unlock)
-
+        sleep(100)
```
Comment (jhugman) [inline app/src/test/java/mozilla/lockbox/store/DataStoreTest.kt]: Ghetto.

File: app/src/main/java/mozilla/lockbox/store/DataStore.kt [hunk 4]
```diff
@@ -50,7 +54,7 @@ open class DataStore(
         state.subscribe { state ->
             when (state) {
                 is State.Locked -> clearList()
-                is State.Unlocked -> updateList()
+                is State.Unlocked -> sync()
```
[the remark this answers is not in the document]
Comment (sashei) [reply app/src/main/java/mozilla/lockbox/store/DataStore.kt]: yeahhh that's a good point; the issue I was running into is that on first-time login, we unlock and subsequently update the list but haven't requested a sync yet. definitely open to alternative ways to wire these things together.

File: app/src/test/java/mozilla/lockbox/store/DataStoreTest.kt (hunk 3 above)
Comment (sashei) [reply app/src/test/java/mozilla/lockbox/store/DataStoreTest.kt]: :(

**Pull request merged**